angular/projects/admin-nrpti/src/app/communications/lng-map-info/lng-map-info.component.ts

```diff
@@ -0,0 +1,148 @@
+})
+export class LngMapInfoComponent implements OnInit, OnDestroy {
+
+  private ngUnsubscribe: Subject<boolean> = new Subject<boolean>();
+
+  public loading = true;
+  public mapForm: FormGroup;
+  public currentSection = null;
+  public lngMapData = [];
+
+  public sections = Constants.LngSectionPickList;
+
+  public tinyMceSettings = {
+    base_url: '/tinymce',
+    suffix: '.min',
+    browser_spellcheck: true,
+    height: 240,
+    plugins: ['lists, advlist, link'],
+    toolbar: [ 'undo redo | formatselect | ' +
+    ' bold italic backcolor | alignleft aligncenter ' +
+    ' alignright alignjustify | bullist numlist outdent indent |' +
+    ' removeformat | help' ]
+  };
+
+  constructor(
+    public route: ActivatedRoute,
+    private factoryService: FactoryService,
+    private loadingScreenService: LoadingScreenService,
+    private _changeDetectionRef: ChangeDetectorRef
+  ) { }
+
+  ngOnInit() {
+    this.route.data.pipe(takeUntil(this.ngUnsubscribe)).subscribe((res: any) => {
+      if (res && res.lngMapData && res.lngMapData[0] && res.lngMapData[0].data) {
+        this.lngMapData = res.lngMapData[0].data.searchResults;
+      } else {
+        alert('Error: could not load LNG Map data');
+      }
+    });
+    this.currentSection = this.lngMapData[0];
+    this.buildForm();
+  }
+
+  checkRoles() {
+    if (this.factoryService.userInRole('sysadmin')) {
+      return true;
+    } else if (this.factoryService.userInLngRole()) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  /**
+   * When selecting a segment, update the form with it's corresponding data
+   *
+   * @param {object} event
+   * @returns
+   * @memberof LngMapInfoComponent
+   */
+  onSelectSegment(event) {
+    if (!event) {
+      return;
+    }
+    const segment = event.target.value.split(': ');
+    this.currentSection = this.lngMapData.find(elem => elem.segment === segment[1]);
+    this.buildForm();
+    this._changeDetectionRef.detectChanges();
+  }
+
+  private buildForm() {
+    this.mapForm = new FormGroup({
+      sectionNumber: new FormControl({
+        value: (this.currentSection && this.currentSection.segment) || '',
+        disabled: !this.checkRoles()
+      }),
+      location: new FormControl({
+        value: (this.currentSection && this.currentSection.location) || '',
+        disabled: !this.checkRoles()
+      }),
+      length: new FormControl({
+        value: (this.currentSection && this.currentSection.length) || '',
+        disabled: !this.checkRoles()
+      }),
+      recentUpdates: new FormControl({
+        value: (this.currentSection && this.currentSection.description) || '',
+        disabled: !this.checkRoles()
+      })
+    });
+  }
+
+  async submit() {
+    this.loadingScreenService.setLoadingState(true, 'main');
+
+    if (
+      !this.mapForm.get('location').value ||
+      !this.mapForm.get('length').value ||
+      !this.mapForm.get('recentUpdates').value
```

Comment:
I'm not sure we should check `recentUpdates` value here.  Currently it doesn't allow user to set an empty value if they want to remove an update.